File: 61-rotate-list/rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        ListNode* H=head;
        if(head==NULL) return NULL;
        if(head->next==NULL) return head;
        int len=0;
        while(head){
            head=head->next;
            len++;
        }
        k=k%len;
        while(k--){
            ListNode* temp=H;
            while(temp->next->next!=NULL){
                temp=temp->next;
            }
            temp->next->next=H;
            H=temp->next;
            temp->next=NULL;
           
        }
        return H;
    }
};

File: 61-rotate-list/rotate-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode RotateRight(ListNode head, int k) {
        if(head==null || head.next==null) return head;
        
        int size=0;
        ListNode temp=head;
        while(temp!=null){
            temp=temp.next;
            size++;
        }
        k=k%size;
        if(k==0) return head;
        int times=size-k;
        temp=head;
        while(temp!=null && times!=1){
            temp=temp.next;
            times--;
        }
        
        ListNode newHead=temp.next;
        temp.next=null;
        temp=newHead;
        while(temp.next!=null){
            temp=temp.next;
        }
        temp.next=head;
        return newHead;
    }
}